src/components/DisplayArticlesByTopic.jsx

```diff
@@ -0,0 +1,48 @@
+import { useParams } from "react-router";
+import { useEffect, useState } from "react";
+import * as api from '../utils/api'
+import moment from "moment";
+import { Link } from "react-router-dom";
+
+function DisplayArticlesByTopic() {
+    const {topic} = useParams()
+    const [articles, setArticles] = useState([]);
+    const [isLoading, setIsLoading] = useState(true)
+
+    useEffect(() => {
+        async function fetchArticles(limit) {
+            let request = await api.getArticles(limit);
+            setIsLoading(false)
+            setArticles(request.filter(article => article.topic === topic));

```

Comment:
Please use the APII request to fetch articles by topic. Filter method should not be used here.